spec/requests/registrations_spec.rb

```diff
@@ -154,6 +175,34 @@
       end
     end
 
+    context "when site configured to accept email registration AND require captcha" do
+      before do
+        allow(SiteConfig).to receive(:allow_email_password_registration).and_return(true)
+        allow(SiteConfig).to receive(:require_captcha_for_email_password_registration).and_return(true)
+      end
+
+      it "creates user when valid params passed and recaptcha completed" do
+        mock_recaptcha_verification
+        post "/users", params:
+          { user: { name: "test #{rand(10)}",
+                    username: "haha_#{rand(10)}",
+                    email: "yoooo#{rand(100)}@yo.co",
+                    password: "PaSSw0rd_yo000",
+                    password_confirmation: "PaSSw0rd_yo000" } }
+        expect(User.all.size).to be 1
+      end
+
+      it "does not create user when valid params passed BUT recaptcha incomplete" do
+        post "/users", params:
+          { user: { name: "test #{rand(10)}",
+                    username: "haha_#{rand(10)}",
+                    email: "yoooo#{rand(100)}@yo.co",
+                    password: "PaSSw0rd_yo000",
+                    password_confirmation: "PaSSw0rd_yo000" } }
+        expect(User.all.size).to be 0
+      end
+    end
```

Comment:
**Suggestion** Test to make sure the appropriate flash/error message is displayed to the user.